### PR TITLE
fix(dialog): mdc-web v1 upgrade

### PR DESCRIPTION
**Pull request by gugu**
Related #697

### CODE REVIEW

File: packages/dialog/index.tsx [hunk 1]
```diff
@@ -136,15 +136,15 @@ class Dialog<T extends HTMLElement = HTMLElement> extends React.Component<
     const {open, autoStackButtons, escapeKeyAction, scrimClickAction} = this.props;
 
     if (prevProps.autoStackButtons !== autoStackButtons) {
-      this.foundation.setAutoStackButtons(autoStackButtons);
+      this.foundation.setAutoStackButtons(autoStackButtons!);
     }
 
     if (prevProps.escapeKeyAction !== escapeKeyAction) {
-      this.foundation.setEscapeKeyAction(escapeKeyAction);
+      this.foundation.setEscapeKeyAction(escapeKeyAction!);
```
Comment (moog16) [inline packages/dialog/index.tsx]: `escapeKeyAction` doesn't have a defaultProp...We can't use `!` here.

File: packages/dialog/index.tsx [hunk 2]
```diff
@@ -136,15 +136,15 @@ class Dialog<T extends HTMLElement = HTMLElement> extends React.Component<
...
     }
 
     if (prevProps.scrimClickAction !== scrimClickAction) {
-      this.foundation.setScrimClickAction(scrimClickAction);
+      this.foundation.setScrimClickAction(scrimClickAction!);
```
Comment (moog16) [inline packages/dialog/index.tsx]: Same with scrimClickAction...we can't use `!`

File: packages/dialog/package.json [hunk 3]
```diff
@@ -11,8 +11,8 @@
     "dialog"
   ],
   "dependencies": {
-    "@material/dialog": "0.43.0",
-    "@material/dom": "^0.41.0",
+    "@material/dialog": "1.0.1",
```
Comment (moog16) [inline packages/dialog/package.json]: can you update these to 1.1.0. Might also need to do that in the other components.

File: package.json [hunk 4]
```diff
@@ -66,7 +66,7 @@
     "@material/card": "^0.41.0",
     "@material/checkbox": "^0.41.0",
     "@material/chips": "^1.0.0",
-    "@material/dialog": "^0.43.0",
+    "@material/dialog": "^1.0.0",
```
Comment (moog16) [inline package.json]: update to v1.1.0

File: packages/dialog/index.tsx [hunk 5]
```diff
@@ -60,15 +61,15 @@ export interface DialogProps<T,
     | React.ReactElement<TitleProps>
   );
   className?: string;
-  escapeKeyAction?: string;
+  escapeKeyAction: string;
```
Comment (moog16) [inline packages/dialog/index.tsx]: make this optional ?

File: packages/dialog/index.tsx [hunk 6]
```diff
@@ -60,15 +61,15 @@ export interface DialogProps<T,
...
   id?: string;
   onClose?: (action: string) => void;
   onClosing?: (action: string) => void;
   onOpen?: () => void;
   onOpening?: () => void;
   open?: boolean;
   role?: 'alertdialog' | 'dialog';
-  scrimClickAction?: string;
+  scrimClickAction: string;
```
Comment (moog16) [inline packages/dialog/index.tsx]: optional

File: packages/dialog/index.tsx [hunk 7]
```diff
@@ -103,6 +104,8 @@ class Dialog<T extends HTMLElement = HTMLElement> extends React.Component<
     id: 'mdc-dialog',
     open: false,
     role: 'alertdialog',
+    escapeKeyAction: strings.CLOSE_ACTION,
```
Comment (moog16) [inline packages/dialog/index.tsx]: Can you document these in the Readme, and also document the defaults.

Comment (gugu) [reply packages/dialog/index.tsx]: Defaults are already written in readme. Do I need to add anything extra?

Comment (moog16) [reply packages/dialog/index.tsx]: Looks good!

**Pull request merged**